pandas/tests/groupby/test_reductions.py

```diff
@@ -1520,3 +1520,27 @@ def test_groupby_std_datetimelike():
     exp_ser = Series([td1 * 2, td1, td1, td1, td4], index=np.arange(5))
     expected = DataFrame({"A": exp_ser, "B": exp_ser, "C": exp_ser})
     tm.assert_frame_equal(result, expected)
+
+def test_mean_numeric_only_validates_bool():
+    """
+    Test that numeric_only parameter only accepts boolean values.
+    See GH#62778
+    """
+    df = pd.DataFrame({"A": range(5), "B": range(5)})
+
+    # These test cases should raise a ValueError
+    msg = "numeric_only accepts only Boolean values"
+    with pytest.raises(ValueError, match=msg):
+        df.groupby(["A"]).mean(["B"])
+
+    with pytest.raises(ValueError, match=msg):
+        df.groupby(["A"]).mean(numeric_only="True")
+
+    with pytest.raises(ValueError, match=msg):
+        df.groupby(["A"]).mean(numeric_only=1)
+
+    # These test cases should work absolutely fine
+    df.groupby(["A"]).mean()
+    df.groupby(["A"]).mean(numeric_only=True)
+    df.groupby(["A"]).mean(numeric_only=False)
```

Comment:
Can you remove these; the test suite has many tests for numeric_only being specified or not specified, these are not increasing our test coverage.